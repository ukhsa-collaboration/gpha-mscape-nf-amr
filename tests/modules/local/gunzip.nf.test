nextflow_process {

    name "Test Process GZ_TO_FASTQ"
    script "modules/local/gunzip.nf"
    process "GZ_TO_FASTQ"

    test("Should run gunzip without any failures") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
            }
            process {
                """
                // define inputs of the process here. Example:
                input[0] = Channel.of(
                    tuple(
                        "climb-id",
                        file("${projectDir}/tests/test_data/ERR3077524_GCF_013349085.1_1x.fq.gz")
                    )
                )
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
