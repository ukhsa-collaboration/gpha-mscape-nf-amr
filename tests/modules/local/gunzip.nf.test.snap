{
    "Should run gunzip without any failures": {
        "content": [
            {
                "0": [
                    [
                        "climb-id",
                        "climb-id.fastq:md5,d04c18908ad4f8aaeda24239a9bf5dc8"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-10-13T14:08:43.850436762"
    },
    "test_gztofastq_positive": {
        "content": [
            {
                "0": [
                    [
                        "climb-id",
                        "climb-id.fastq:md5,d04c18908ad4f8aaeda24239a9bf5dc8"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-10-13T14:27:53.463831543"
    }
}